new-docs/source/tutorial/2-advanced-execution.ipynb

```diff
@@ -0,0 +1,428 @@
+    "\n",
+    "Pydra caches all task results in the runtime cache (see [File-system locations](./1-getting-started.html##File-system-locations))\n",
+    "as long as exactly the hashes of the inputs provided to the task are the same. Here we\n",
+    "go through some of the practicalities of this caching and hashing (see\n",
+    "[Caches and hashes](../explanation/hashing-caching.html) for more details and issues\n",
+    "to consider)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If we attempt to run the same task with the same parameterisation the cache directory\n",
+    "will point to the same location and the results will be reused"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from copy import copy\n",
+    "from pathlib import Path\n",
+    "import tempfile\n",
+    "from pprint import pprint\n",
+    "from fileformats.medimage import Nifti1\n",
+    "from pydra.engine.submitter import Submitter\n",
+    "from pydra.tasks.mrtrix3.v3_0 import MrGrid\n",
+    "\n",
+    "# Make a temporary directory\n",
+    "test_dir = Path(tempfile.mkdtemp())\n",
+    "nifti_dir = test_dir / \"nifti\"\n",
+    "nifti_dir.mkdir()\n",
+    "\n",
+    "# Generate some random NIfTI files to work with\n",
+    "nifti_files = [Nifti1.sample(nifti_dir, seed=i) for i in range(10)]\n",
+    "\n",
+    "VOX_SIZES = [\n",
+    "        (0.5, 0.5, 0.5),\n",
+    "        (0.25, 0.25, 0.25),\n",
+    "        (0.1, 0.1, 0.1),\n",
+    "        (0.35, 0.35, 0.35),\n",
+    "        (0.1, 0.1, 0.1),\n",
+    "        (0.5, 0.5, 0.5),\n",
+    "        (0.25, 0.25, 0.25),\n",
+    "        (0.2, 0.2, 0.2),\n",
+    "        (0.35, 0.35, 0.35),\n",
+    "        (0.1, 0.1, 0.1),\n",
+    "    ]\n",
+    "\n",
+    "mrgrid_varying_vox = MrGrid(operation=\"regrid\").split(\n",
+    "    (\"in_file\", \"voxel\"),\n",
+    "    in_file=nifti_files,\n",
+    "    voxel=VOX_SIZES,\n",
+    ")\n",
+    "\n",
+    "submitter = Submitter(cache_dir=test_dir / \"cache\")\n",
+    "\n",
+    "\n",
+    "with submitter:\n",
+    "    result1 = submitter(mrgrid_varying_vox)\n",
+    "\n",
+    "\n",
+    "mrgrid_varying_vox2 = MrGrid(operation=\"regrid\").split(\n",
+    "    (\"in_file\", \"voxel\"),\n",
+    "    in_file=nifti_files,\n",
+    "    voxel=copy(VOX_SIZES),\n",
+    ")\n",
+    "\n",
+    "# Result from previous run is reused as the task and inputs are identical\n",
+    "with submitter:\n",
+    "    result2 = submitter(mrgrid_varying_vox2)\n",
+    "\n",
+    "# Check that the output directory is the same for both runs\n",
+    "assert result2.output_dir == result1.output_dir\n",
+    "\n",
+    "# Change the voxel sizes to resample the NIfTI files to for one of the files\n",
+    "mrgrid_varying_vox2.voxel[2] = [0.25]\n",
+    "\n",
+    "# Result from previous run is reused as the task and inputs are identical\n",
```

Comment:
that sentence probably should be changed. However I don't remember now if pydra still collects the old results for elements of the splits that have not changed the voxel value

Reply:
If I understand what you mean old results should be collected shouldn't they? Because they will be stored in a separate "job" directory, which should match the hash of subsequent jobs

Reply:
yes, I would say that most of the results should be collected from previous jobs. Just want to confirm, since the description was focused on the main directory.